### PR TITLE
Add ActiveWorkspaceProjectContextTracker

### PR DESCRIPTION
**Pull request by davkean**
This tracks Roslyn's active context by handling VSHPROPID_ActiveIntellisenseProjectContext - consumption in a future PR.

### CODE REVIEW

**Comment by davkean** [comment]
cc @jasonmalinowski

**Comment by davkean** [comment]
This will replace the need to have these types:

https://github.com/dotnet/project-system/blob/master/src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/IConfiguredProjectHostObject.cs
https://github.com/dotnet/project-system/blob/master/src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/IUnconfiguredProjectHostObject.cs
https://github.com/dotnet/project-system/blob/master/src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/LanguageServices/AbstractHostObject.cs
https://github.com/dotnet/project-system/blob/master/src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/LanguageServices/ConfiguredProjectHostObject.cs
https://github.com/dotnet/project-system/blob/master/src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/LanguageServices/UnconfiguredProjectHostObject.cs

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/LanguageServices/ActiveWorkspaceProjectContextTracker.cs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+        [ImportingConstructor]
+        public ActiveWorkspaceProjectContextTracker(UnconfiguredProject project) // For scoping
+        {
+        }
+
+        public string ActiveIntellisenseProjectContext
+        {
+            get;
+            set;
```
Comment (jmarolf) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/LanguageServices/ActiveWorkspaceProjectContextTracker.cs]: What are the threading rules about setting this value?

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS.UnitTests/ProjectSystem/VS/LanguageServices/ActiveWorkspaceProjectContextTrackerTests.cs [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+            var context = IWorkspaceProjectContextMockFactory.Create();
+
+            Assert.Throws<ArgumentNullException>("contextId", () =>
+            {
+                instance.RegisterContext(context, (string)null);
+            });
+        }
+
+        [Fact]
+        public void RegisterContext_EmptyAsContextId_ThrowsArgumentNull()
```
Comment (Pilchie) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS.UnitTests/ProjectSystem/VS/LanguageServices/ActiveWorkspaceProjectContextTrackerTests.cs]: `ThrowsArgument`?

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/LanguageServices/ActiveWorkspaceProjectContextTracker.cs (hunk 1 above)
Comment (davkean) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/LanguageServices/ActiveWorkspaceProjectContextTracker.cs]: It's always set on UI thread, but if it's not what concerns do you have? Being out of sync after we've queried it?

**Pull request merged**